Improvements to nonscaffolded Prim and Dijkstra

ttaiv opened the pull request:
Formatted both exercises and updated their instructions. Ensured the uniqueness of edge weights in Prim and increased the interval of edge weights to be [10, 90). This allowed to remove the requirement for processing nodes at equal distances in alphabetic order. Also updated the pseudo code of the Prim exercise.

atilante reviewed (changes requested):
# Manual testing

PrimAVPE

- [x] Edge weights are at interval [10,90]
- [x] Discarded alphabetic order requirement from instructions
- [x] Pseudocode updated accordingly
- [x] English and Finnish instructions match

DijkstraPE-research

- [x] Exercise instructions elaborated
- [x] English and Finnish instructions match

DijkstraPE-research-v3

- No new/changed functionality
- [x] Exercise starts
- [x] Model answer is generated

# Code review

PrimAVPE

- [x] Edge weights are unique in (function `hasUniqueEdgeWeights` in
  `graphUtils.js`)
- [x] Edge weights are at interval [10,90] (constants `MIN_EDGE_WEIGHT` and
  `MAX_EDGE_WEIGHT` in `PrimAVPE.js`)

DijkstraPE-research

- [x] Exercise instructions elaborated
- [x] English and Finnish instructions match

- code reformatting
- variable rename: `nodeIndex` -> `minIndex`

DijkstraPE-research-v3

- Function `researchInstanceToAdjList()` relocated

# Comments

1. Nice set implementation at `hasUniqueEdgeWeights()`!

2. I remember vaguely that the `visited` flag in PrimAVPE was added to make it
   consistent with other graph algorithm exercises. However, DijkstraPE-research
   does not have that flag. This inconsistency should be addressed.

3. Function `researchInstanceToAdjList()` was moved from
   `DijkstraPE-research-v3.js` refactored into `graphUtils.js`. Please move this
   back. The function is only used at DijkstraPE-research-v3 and
   it only processes data produced by method `generateInstance()` at `DijkstraInstanceGenerator.js`.

ttaiv commented:
Replies to comments above:

1. Thanks! :)
2. Actually I added the visited flag to the pseudo code primarily to make the pseudo code consistent with the one currently used in the course material (the visited flag was added to it recently). But it is true that it could also be added to the DijkstraPE-research exercise (even though to my understanding it is not needed to make the pseudo code correct as it is with Prim).
3. A good point. I originally moved the function to reuse it for creating adjacency list for DijkstraPE-research but we changed the plans and did not add it to the exercise. I reverted the change.

atilante commented:
As discussed in Archie's DSA course memo 29.10.2024

Prim
    v → vertex in pseudocode tehty
    (u, previous[u]) to exercise instructions

These have been done.

Check: function researchInstanceToAdjList was moved back to DijkstraPE-research-v3.js from graphUtils.js.

Approved.
